addons/print/tests/common.py

```diff
@@ -11,85 +11,90 @@
         cls.printer_default.set_system_default()
 
+        # Set the context variable to true so that ir.actions.report.render_qweb_pdf() actually attempt to generate a pdf
+        cls.printer_default = cls.printer_default.with_context(force_report_rendering=True)
+
         # Locate test file directory corresponding to the class (which
         # may be a derived class in a different module).
         module_file = sys.modules[cls.__module__].__file__
         module = get_resource_from_path(module_file)[0]
 
-        path = get_resource_path(module, 'tests', 'files')
+        path = get_resource_path(module, "tests", "files")
         if path:
             cls.files = pathlib.Path(path)
 
         cls.safety = "print.default_test_print"
         # Enable default print safety for tests
         config.misc["print"] = {"default_test_print": 1}
 
+        # Configure the static files before trying to access them.
+        # This will avoid recieving WARNING messages from the logger
+        # when it goes to request the static files only to find they haven't
+        # been configured.
+        root.load_addons()
+
     def setUp(self):
         super().setUp()
 
         # Patch find_in_path() as used in print_printer.py
         patch_find_in_path = patch(
-            'odoo.addons.print.models.print_printer.find_in_path',
-            autospec=True, return_value=MOCK_LPR,
+            "odoo.addons.print.models.print_printer.find_in_path",
+            autospec=True,
+            return_value=MOCK_LPR,
         )
         self.mock_find_in_path = patch_find_in_path.start()
         self.addCleanup(patch_find_in_path.stop)
 
         # Patch subprocess as used in print_printer.py
         patch_subprocess = patch(
-            'odoo.addons.print.models.print_printer.subprocess',
+            "odoo.addons.print.models.print_printer.subprocess",
             autospec=True,
         )
         self.mock_subprocess = patch_subprocess.start()
         self.addCleanup(patch_subprocess.stop)
 
         # Create mock lpr subprocess
         self.mock_lpr = Mock()
-        self.mock_lpr.communicate.return_value = ('', '')
+        self.mock_lpr.communicate.return_value = ("", "")
         self.mock_lpr.returncode = 0
         self.mock_subprocess.Popen.return_value = self.mock_lpr
 
         # Force test_enable to True (which is not necessarily the case
         # when tests are run via the "-f" command-line option) to
         # prevent ir.actions.report from committing the assets bundle
         # and hence releasing the savepoint.
-        #
-        # Create mock test_report_directory to ensure that
-        # ir.actions.report.render_qweb_pdf() will actually attempt to
-        # generate a PDF
-        #
-        self.tempdir = tempfile.TemporaryDirectory()
-        self.addCleanup(self.tempdir.cleanup)
-        patch_config = patch.dict(config.options, {
-            'test_enable': True,
-            'test_report_directory': self.tempdir.name,
```

Comment:
Q: Why isn't this needed anymore?

Reply:
The test_enable patch config is still needed. The temporary file is not. They were put in 11 as when you try to render a pdf it will return html if you do not have those two config parameters (https://github.com/unipartdigital/odoo/blob/udes-11.0/odoo/addons/base/ir/ir_actions_report.py#:~:text=if%20tools.config%5B'test_enable'%5D%20and%20not%20tools.config%5B'test_report_directory'%5D%3A).

In 14 this has been changed, such that a pdf won’t be rendered if there is not a special context set (https://github.com/unipartdigital/odoo/blob/udes-14.0/odoo/addons/base/models/ir_actions_report.py#:~:text=if%20(tools.config%5B'test_enable'%5D%20or%20tools.config%5B'test_file'%5D)%20and%20not%20self.env.context.get('force_report_rendering')%3A). This meant I had to add the with_context variable to some printers/report environments in the tests.

Reply:
Sorry I should have been more clear, I was referring to `test_report_directory`. But the code you shared shows that, thanks for clarifying. :)